src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_params.py

```diff
@@ -0,0 +1,97 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands.parameters import (
+    resource_group_name_type, get_resource_group_completion_list)
+
+from azure.cli.command_modules.resource._completers import (
+    get_policy_set_completion_list, get_policy_completion_list,
+    get_policy_assignment_completion_list, get_providers_completion_list, get_resource_types_completion_list)
+
+from ._validators import validate_resource
+
+
+def load_arguments(self, _):
+    for scope in ['state', 'event']:
+        with self.argument_context('policy {}'.format(scope)) as c:
+            c.argument(
+                'management_group_name',
+                options_list=['--management-group-name', '-m'],
+                help='Management group name.')
+            c.argument(
+                'resource_group_name',
+                options_list=['--resource-group-name', '-g'],
+                arg_type=resource_group_name_type,
+                completer=get_resource_group_completion_list,
+                help='Resource group name.')
+            c.argument(
+                'resource',
+                options_list=['--resource', '-r'],
+                validator=validate_resource,
+                help='Resource id or resource name.')
+            c.argument(
+                'resource_provider_namespace',
+                options_list=['--resource-provider-namespace', '-n'],
+                completer=get_providers_completion_list,
+                help='Resource provider namespace.')
+            c.argument(
+                'resource_type',
+                options_list=['--resource-type', '-t'],
+                completer=get_resource_types_completion_list,
+                help='Resource type.')
```

Comment:
Can you format the resource args to align more closely with that of `az resource`?
Help for `resource show` shown below:
```
(env36) C:\Users\wilx\azure-cli>az resource show -h

Command
    az resource show: Get the details of a resource.

Arguments
    --include-response-body: Use if the default command output doesn't capture all of the property
                             data.  Allowed values: false, true.

Resource Id Arguments
    --api-version          : The api version of the resource (omit for latest).
    --ids                  : One or more resource IDs (space-delimited). If provided, no other
                             "Resource Id" arguments should be specified.
    --name -n              : The resource name. (Ex: myC).
    --namespace            : Provider namespace (Ex: 'Microsoft.Provider').
    --parent               : The parent path (Ex: 'resA/myA/resB/myB').
    --resource-group -g    : Name of resource group. You can configure the default group using `az
                             configure --defaults group=<name>`.
    --resource-type        : The resource type (Ex: 'resC'). Can also accept namespace/type format
                             (Ex: 'Microsoft.Provider/resC').
```

Reply:
Look into the `id_part` kwarg

Reply:
I will make the changes to align other resource parameters with the resources command set (will change names and add a parent parameter)

Reply:
Yep, ignore the `--ids` parameter and allow `--resource` to accept both an id and name.